plugins/discovery-ec2/src/main/java/org/elasticsearch/discovery/ec2/AwsEc2UnicastHostsProvider.java

```diff
@@ -113,8 +116,19 @@ public AwsEc2UnicastHostsProvider(Settings settings, TransportService transportS
             // NOTE: we don't filter by security group during the describe instances request for two reasons:
             // 1. differences in VPCs require different parameters during query (ID vs Name)
             // 2. We want to use two different strategies: (all security groups vs. any security groups)
-            descInstances = client.describeInstances(buildDescribeInstancesRequest());
+            SecurityManager sm = System.getSecurityManager();
+            if (sm != null) {
+                // unprivileged code such as scripts do not have SpecialPermission
+                sm.checkPermission(new SpecialPermission());
+            }
+            descInstances = AccessController.doPrivileged(new PrivilegedAction<DescribeInstancesResult>() {
+                @Override
+                public DescribeInstancesResult run() {
+                    return client.describeInstances(buildDescribeInstancesRequest());
+                }
+            });
         } catch (AmazonClientException e) {
+            logger.error("Exception while retrieving instance list from AWS API", e);
```

Comment:
In case of an error, we will print 2 times a similar line if info is enabled? Shouldn't we remove the logger.info and logger.debug given that they don't give more insight?